plenoptic/metric/perceptual_distance.py

```diff
@@ -56,9 +57,9 @@ def _ssim_parts(img1, img2, dynamic_range, pad=False):
                           f" img1: {img_ranges[0]}, img2: {img_ranges[1]}. "
                           "Continuing anyway...")
     (n_batches, n_channels, height, width) = img1.shape
-    if n_channels > 1:
-        warnings.warn("SSIM was developed on grayscale images, no guarantee "
-                      "it will make sense for more than one channel!")
+    if n_channels > 1 or img2.shape[1] > 1:
+        warnings.warn("SSIM was developed on grayscale images, so the channel dimension "
+                      "is treated as another batch dimension.")
     if img2.shape[-2:] != (height, width):
```

Comment:
do `img1` and `img2` have to have the same number of channels (if they don't have a single channel)? If I'm reading this correctly, they can either have the same number of batches (in which case they're compared batch-wise) or one of them can have a single batch -- do we want something similar for channels?

Reply:
Yes, both batch and channel dimensions follow the broadcast rule.